core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceMultipartTest.java

```diff
@@ -89,6 +90,31 @@ void testUploadFile(String path) throws Exception {
                            "\"param1\":\"armeria\"}");
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = "/uploadWithUnintendedFileParam")
+    void testUploadFileWithUnexpectedParameters(String path) throws Exception {
+        final Multipart multipart = Multipart.of(
+                BodyPart.of(ContentDisposition.of("form-data", "file1", "foo.txt"), "foo"),
+                BodyPart.of(ContentDisposition.of("form-data", "path1", "bar.txt"), "bar"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile1", "qux.txt"), "qux"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile2", "quz.txt"),
+                            MediaType.PLAIN_TEXT, "quz"),
+                // Unexpected parameter: multipartFile3
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile3", "qux3.txt"), "qux3"),
+                BodyPart.of(ContentDisposition.of("form-data", "param1"), "armeria")
+
+        );
+        final AggregatedHttpResponse response =
+                server.blockingWebClient().execute(multipart.toHttpRequest(path));
+        assertEquals(HttpStatus.OK, response.status());
```

Comment:
We prefer assertJ for richer error messages so that we can debug flaky tests more easily

```suggestion
        assertThat(HttpStatus.OK).isEqualTo(response.status());
```